### PR TITLE
Fixed #386: Enrichment tag upload - copying uploaded tags into the final table

### LINKED ISSUE #386: Enrichment tag upload - copying uploaded tags into the final table
The uploaded enrichment tags should be copied after the transmission into the final EnrichmentTag table. The best idea for now seems to be doing in one transaction the following operations:
- deleting all the records from the final table,
- copying the newly uploaded tags into it.

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/UploadEnrichmentTagRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    private UploadEnrichmentTagRepository uploadEnrichmentTagRepository;
+    
+   
+    
+    @Before
+    public void before() {
+        
+        
+    }
+
```
Comment (pawelczak) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/UploadEnrichmentTagRepositoryTest.java]: Empty method

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public final class TestInMemoryUploadEnrichmentTagFactory {
+   
+    
+    private TestInMemoryUploadEnrichmentTagFactory() {
+        
+        throw new IllegalStateException("may not be instantiated");
+        
+    }
+    
+
```
Comment (pawelczak) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java]: Missing class section logic

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import pl.edu.icm.saos.persistence.enrichment.model.UploadEnrichmentTag;
+
+/**
+ * @author Łukasz Dumiszewski
+ */
+
+public final class TestInMemoryUploadEnrichmentTagFactory {
+   
+
```
Comment (pawelczak) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java]: Missing class section constructors

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/UploadEnrichmentTagRepositoryTest.java (hunk 1 above)
Comment (lukdumi) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/UploadEnrichmentTagRepositoryTest.java]: :+1:

Comment (lukdumi) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/UploadEnrichmentTagRepositoryTest.java]: :+1:

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/EnrichmentTagRepositoryTest.java [hunk 4]
```diff
@@ -26,23 +32,24 @@
...
     
     @Before
     public void before() {
         
-        enrichmentTag.setTagType(EnrichmentTagTypes.REFERENCED_CASE_NUMBERS);
-        enrichmentTag.setValue("{\"caseNumbers\": [\"123\", \"234\"]}");
         
     }
-    
+
```
Comment (pawelczak) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/EnrichmentTagRepositoryTest.java]: Empty method

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java (hunk 2 above)
Comment (lukdumi) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java]: :+1:

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/EnrichmentTagRepositoryTest.java (hunk 4 above)
Comment (lukdumi) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/EnrichmentTagRepositoryTest.java]: :+1:

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java (hunk 3 above)
Comment (lukdumi) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/enrichment/TestInMemoryUploadEnrichmentTagFactory.java]: :+1:

File: saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/process/UploadEnrichmentTagOverwriterTest.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+        // execute
+        
+        uploadEnrichmentTagOverwriter.overwriteEnrichmentTags();
+        
+        // assert
+        
+        verify(enrichmentTagRepository).deleteAll();
+        verify(uploadEnrichmentTagCopier).copyUploadedEnrichmentTags();
+        verify(uploadEnrichmentTagRepository, never()).deleteAll();
```
Comment (pawelczak) [inline saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/process/UploadEnrichmentTagOverwriterTest.java]: Why you check here for `verify(uploadEnrichmentTagRepository, never()).deleteAll();`? Is it a precaution, that in the future no one should try to remove uploadedEnrichmentTags after they are copied? Or is it just to show that uploadedET are not removed after copying process?

Comment (lukdumi) [inline saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/process/UploadEnrichmentTagOverwriterTest.java]: A mistake. Should be: verifyZeroInteractions(uploadEnrichmentTagRepository);

**Pull request closed without merging**